test/run_test.py

```diff
@@ -340,6 +349,82 @@ def print_to_stderr(message):
     print(message, file=sys.stderr)
 
 
+def get_test_time_reports_from_S3():
+    commit_date_str = subprocess.check_output(
+        ['git', 'show', '-s', '--format=%ci', 'HEAD'],
+        encoding="ascii").splitlines()[0]
+    commit_date = datetime.strptime(commit_date_str, '%Y-%m-%d %H:%M:%S %z')
+    day_before_commit = str(commit_date - timedelta(days=1)).split(' ')[0]
+    # something like git rev-list --before="2021-02-24" --max-count=1 --remotes="*upstream/nightly"
+    nightly_commit = subprocess.check_output(
+        ["git", "rev-list", f"--before={day_before_commit}", "--max-count=1", "--remotes=*upstream/nightly"],
+        encoding="ascii").splitlines()[0]
+
+    job = os.environ.get("CIRCLE_JOB", "")
+    job_minus_shard_number = job.rstrip('0123456789')
+    s3 = boto3.resource("s3", config=botocore.config.Config(signature_version=botocore.UNSIGNED))
```

Comment:
Please avoid throwing an exception of BOTO is not available
```suggestion
    if not HAVE_BOTO3:
        print("Give me BOTO and I will shard tests for you!") 
        return []
    s3 = boto3.resource("s3", config=botocore.config.Config(signature_version=botocore.UNSIGNED))
```